.github/workflows/firebase-preview.yml

```
@@ -20,13 +24,67 @@ jobs:
+          echo "GOOGLE_API_KEY=$API_KEY" >> $GITHUB_ENV
+          # Create .env.production for Firebase deployment (the working approach)
+          echo "GOOGLE_API_KEY=$API_KEY" > .env.production
+          rm /tmp/service-account.json
+      
+      - name: Build project  
         run: npm run build
+        env:
+          GOOGLE_API_KEY: ${{ env.GOOGLE_API_KEY }}
+          NEXT_PUBLIC_BUILD_COMMIT: ${{ github.sha }}
+          NEXT_PUBLIC_BUILD_BRANCH: ${{ github.head_ref }}
+          NEXT_PUBLIC_BUILD_TIME: ${{ github.event.head_commit.timestamp || github.run_started_at }}
```

actionlint (1.7.7)

property "run_started_at" is not defined in object type {action: string; action_path: string; action_ref: string; action_repository: string; action_status: string; actor: string; actor_id: string; api_url: string; artifact_cache_size_limit: number; base_ref: string; env: string; event: object; event_name: string; event_path: string; graphql_url: string; head_ref: string; job: string; output: string; path: string; ref: string; ref_name: string; ref_protected: bool; ref_type: string; repository: string; repository_id: string; repository_owner: string; repository_owner_id: string; repository_visibility: string; repositoryurl: string; retention_days: number; run_attempt: string; run_id: string; run_number: string; secret_source: string; server_url: string; sha: string; state: string; step_summary: string; token: string; triggering_actor: string; workflow: string; workflow_ref: string; workflow_sha: string; workspace: string}

(expression)